### ISSUE TITLE
GPU accelerated Open3d seems to take just as much time on segment_plane() 

### ISSUE THREAD

**Issue by pooja-kabra** [isl-org/Open3D#6146]
### Checklist

- [X] I have searched for [similar issues](https://github.com/isl-org/Open3D/issues).
- [X] For Python issues, I have tested with the [latest development wheel](http://www.open3d.org/docs/latest/getting_started.html#development-version-pip).
- [X] I have checked the [release documentation](http://www.open3d.org/docs/release/) and the [latest documentation](http://www.open3d.org/docs/latest/) (for `master` branch).

### My Question

I am trying to accelerate plane fitting in my application by using GPU accelerated Open3D (built from source with CUDA enabled).
````
def fit_plane(xyz, sample_size): 

    pcl_o3d = o3d.geometry.PointCloud()
    pcl_o3d.points = o3d.utility.Vector3dVector(xyz.T)
    # ransac_n =3 might not be enough
    plane_model, inliers = pcl_o3d.segment_plane(distance_threshold=1,
                                            ransac_n=sample_size,
                                            num_iterations=30)
    return plane_model`
````
    
This above is the old version of my function. segment_plane() takes 0.45 s
Below is the new one:
````
def fit_plane(xyz, sample_size):
    
    device = o3d.core.Device('CUDA:0' if o3d.core.cuda.is_available() else 'CPU:0')
    dtype = o3d.core.float32
    pcl_o3d = o3d.t.geometry.PointCloud(device)
    pcl_o3d.point.positions = o3d.core.Tensor(xyz.T, dtype, device)
    plane_model, inliers = pcl_o3d.segment_plane(distance_threshold=1,
                                            ransac_n=sample_size,
                                            num_iterations=30)
    plane_model = plane_model.cpu().numpy()
    return plane_model`
````

Here o3d.core.Tensor() takes 0.37 seconds on top of the 0.46 s taken by segment_plane(). I am trying to understand if there are any flaws in my code because it beats the purpose of using the CUDA installation of the library if it is going to take more time.
Any inputs would be appreciated, thanks!

**Comment by theNded** [comment]
Unfortunately, no CUDA kernels have been implemented for this feature. The tensor interface simply wraps the legacy CPU version: https://github.com/isl-org/Open3D/blob/master/cpp/open3d/t/geometry/PointCloud.cpp#L1111

In general, the 0.37s is an inevitable overhead to initialize the cuda cache and conduct CPU-GPU transfer. Usually after a warm-up this can be reduced.

**Comment by pooja-kabra** [comment]
Thanks for the response @theNded ! What do you mean after a warm-up? Do you mean the next time I make an o3d.core.Tensor call, it will take less time?

**Comment by theNded** [comment]
Yes, the first cuda call should take some time, but the following ones should take less. This also applies to other cuda-dependent packages like torch.

**Comment by pooja-kabra** [comment]
Sounds good, thank you✅
